### DISCUSSION TITLE
ALE fixers and on save not working

### DISCUSSION THREAD

**Discussion by iJKTen** [w0rp/ale#4484] [Q&A - Ask for help with problems]
Hi I have the following in my `.vimrc` located at `~/.vimrc`
```
let g:ale_fix_on_save = 1
let g:ale_linters = {'javascript': ['eslint']}
let g:ale_fixers = {
 \ 'javascript': ['prettier', 'eslint'],
 \ 'typescript': ['prettier', 'eslint'],
 \ 'javascriptreact': ['prettier', 'eslint'],
 \ 'typescriptreact': ['prettier', 'eslint'],
 \ }
```

When I open a JS file containing JSX I do see an error and when I run `:ALEFix` to fix it I see `No fixers have been defined. Try :AleFixSuggest`

Running `set ft` shows `filetype=javascript`

Running `ALEInfo` shows 
```
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
```

so the globals are not getting picked up. 

I am using `nvim` is that matters. 

What am I doing wrong? 

Thank you

**Comment by iJKTen** [comment]
I now have the following 
```
let g:ale_fix_on_save = 1
let g:ale_linters = {'javascript': ['eslint']}
let g:ale_fixers = {
 \ 'javascript': ['prettier', 'eslint'],
 \ 'typescript': ['prettier', 'eslint'],
 \ 'javascriptreact': ['prettier', 'eslint'],
 \ 'typescriptreact': ['prettier', 'eslint'],
 \ }
```

in `~/.vimrc`, `~/.vim/.vimrc`, and `~/.vim/ftplugin/javascript.vim` and running `ALEFix` still says `No fixers have been defined. Try :AleFixSuggest`

Running `ALEInfo` still shows
```
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
```

I am using `neovim`

**Reply by hsanson** [reply]
Neovim usually uses `~/.config/nvim/init.vim` as default config...   read `:h init.vim` for more details.

**Comment by iJKTen** [comment]
Thank you! Got it working!
